env/env_test.cc

```diff
@@ -247,6 +247,40 @@ TEST_F(EnvPosixTest, MemoryMappedFileBuffer) {
   ASSERT_EQ(expected_data, actual_data);
 }
 
+TEST_F(EnvPosixTest, LoadLibrary) {
+  std::shared_ptr<DynamicLibrary> library;
+  std::function<void*(void *, const char*)> function;
+  Status status = env_->LoadLibrary("no-such-library", "", &library);
+  ASSERT_NOK(status);
+  ASSERT_EQ(nullptr, library.get());
+#if ! defined(OS_WIN)
+  status = env_->LoadLibrary("dl", "", &library);
+  ASSERT_OK(status);
+  ASSERT_NE(nullptr, library.get());
+  ASSERT_NE(nullptr, library->LoadFunction("dlsym", &function));
+  ASSERT_NE(nullptr, function);
+  ASSERT_OK(env_->LoadLibrary(library->Name(), "", &library));
+#endif
+}
+
+#if ! defined(OS_WIN)
+TEST_F(EnvPosixTest, LoadLibraryWithSearchPath) {
+  std::shared_ptr<DynamicLibrary> library;
+  std::function<void*(void *, const char*)> function;
+  ASSERT_NOK(env_->LoadLibrary("no-such-library", "/tmp", &library));
+  ASSERT_EQ(nullptr, library.get());
+  ASSERT_NOK(env_->LoadLibrary("dl", "/tmp", &library));
+  ASSERT_EQ(nullptr, library.get());
+
+  ASSERT_OK(env_->LoadLibrary("dl", "/tmp:/usr/lib:/usr/lib64", &library));
```

Comment:
Do we expect to find `dl` in this path on all non-WIN platforms? Did some random search, found this one: https://docs.oracle.com/cd/E26502_01/html/E29033/libdl-3lib.html. I cannot verify because I do not have a Solaris machine.

Reply:
According to the documentation I can find on dlopen, it is implemented in libdl.  Since we are using dlopen to open the library, we must have libdl on the path somewhere, the question is where.  The code at line 277 checks for it in /usr/lib (MacOS) and Linux (usr/lib64) for the boxes I can test on.  I do not know if that is where it will always be found or not.